server/json_client.go

```diff
@@ -0,0 +1,105 @@
+package server
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/mysterium/node/identity"
+	"net/http"
+	"net/url"
+)
+
+const (
+	mysteriumAgentName       = "goclient-v0.1"
+	authenticationHeaderName = "Authorization"
+	authenticationSchemaName = "Signature"
+)
+
+var noSignature = identity.SignatureBytes(nil)
+
+type HttpClient interface {
+	DoGet(path string, values url.Values) (*http.Response, error)
+	DoPost(path string, body interface{}) (*http.Response, error)
+	DoSignedPost(path string, body interface{}, signer identity.Signer) (*http.Response, error)
+}
+
+//HttpTransport interface with single method do is extracted from net/transport.Client structure
+type HttpTransport interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+func NewJsonClient(baseUrl string, transport HttpTransport) *jsonHttpClient {
+	return &jsonHttpClient{
+		baseUrl,
+		transport,
+	}
+}
+
+type jsonHttpClient struct {
+	baseApiUrl string
+	transport  HttpTransport
+}
+
+func (jhc *jsonHttpClient) DoGet(path string, values url.Values) (*http.Response, error) {
+	pathWithQuery := fmt.Sprintf("%v?%v", path, values.Encode())
```

Comment:
Looks like not building "/path", then dont have query params

Reply:
Please explain. The idea is path or endpoint concated it with params. I.e. find proposals:
client.DoGet("proposals" , {map with params to find proposal with node id})

Reply:
I am talking about case, then query params are empty.
`DoGet('v1/path', url.Values{})`
IMHO should be asserted to become:
`/path`